crates/sage/src/enzyme.rs

```diff
@@ -76,6 +82,7 @@ impl std::hash::Hash for Digest {
     fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
         self.sequence.hash(state);
         self.position.hash(state);
+        self.start_position.hash(state);
```

Comment:
I don't think this is correct behavior (won't necessarily cause a bug though). We use `hash` to deduplicate digests prior to creating peptides.

Only `sequence` and `position` are considered to ensure that we don't accidentally deduplicate peptide sequences that occur on protein termini, which might be assigned termini-specific modifications. Including `start_position` will lead to extra (duplicated) digests being created -> more duplicated peptides that must be generated (expensive) and then trashed during duplication.

Reply:
Okay, that makes sense. I was thinking about maybe including a list of start indices for each protein in the `proteins` list, but that probably doesn't make sense from a peptide-centric search perspective, and other search engines just list the index for the leading protein.